<!DOCTYPE html>
<html lang="en" class="no-js">
    <head>
        <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">






<title>Example Analysis</title>
<meta name="description" content="">

<link rel="canonical" href="https://kaizheng-academic.github.io/code/lrtm-2025-tcbb/documentation/04_CARD_Example.html">
<link rel="alternate" type="application/rss+xml" title="" href="/code/lrtm-2025-tcbb/feed.xml">




<link
    rel="stylesheet"
    href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.4.1/css/bootstrap.min.css"
    integrity="sha256-L/W5Wfqfa0sdBNIKN9cG6QA5F2qx4qICmU2VgLruv9Y="
    crossorigin="anonymous"
/>
<link
    rel="stylesheet"
    href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css"
    integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk="
    crossorigin="anonymous"
/>
<link
    rel="stylesheet"
    href="/code/lrtm-2025-tcbb/assets/css/styles.css"
/>



    </head>

    
    <body
        class="layout-page"
        data-spy="scroll"
        data-target="#page-toc"
        data-offset="0"
    >
        <header class="navbar navbar-expand-md navbar-custom" id="top">
    <div class="container">
        <a class="navbar-brand" href="/code/lrtm-2025-tcbb/">CARD</a>

        <button
            class="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
        >
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
        </button>

        <nav class="navbar-collapse collapse" role="navigation" id="navbarSupportedContent">
            <ul class="navbar-nav">
                
                
                    
                    

                    
                    

                    
                
                    
                    

                    
                    

                    
                        <li class="nav-item ">
                            <a
                                class="nav-link"
                                href="/code/lrtm-2025-tcbb/documentation/01_About.html"
                            >
                                About
                            </a>
                        </li>
                    
                
                    
                    

                    
                    

                    
                
                    
                    

                    
                    

                    
                        <li class="nav-item ">
                            <a
                                class="nav-link"
                                href="/code/lrtm-2025-tcbb/documentation/03_data.html"
                            >
                                Data Input
                            </a>
                        </li>
                    
                
                    
                    

                    
                    

                    
                        <li class="nav-item active">
                            <a
                                class="nav-link"
                                href="/code/lrtm-2025-tcbb/documentation/04_CARD_Example.html"
                            >
                                Example Analysis
                            </a>
                        </li>
                    
                
                    
                    

                    
                    

                    
                        <li class="nav-item ">
                            <a
                                class="nav-link"
                                href="/code/lrtm-2025-tcbb/documentation/05_Experiments.html"
                            >
                                Experiments
                            </a>
                        </li>
                    
                
                    
                    

                    
                    

                    
                        <li class="nav-item ">
                            <a
                                class="nav-link"
                                href="/code/lrtm-2025-tcbb/documentation/02_installation.html"
                            >
                                Installation
                            </a>
                        </li>
                    
                
            </ul>
            <ul class="navbar-nav ml-auto">
                
                    <li class="nav-item ">
                        <a
                            class="nav-link"
                            href="https://github.com/YMa-lab/CARD/"
                        >
                            
    <i
        class="fab fa-github fa-fw "
        aria-hidden="true"
    ></i>


                            GitHub
                        </a>
                    </li>
                
            </ul>
        </nav>
    </div>
</header>

        
<div class="site-masthead mb-5 py-5 text-left">
    <div class="container">
        <h1 class="site-masthead__title mb-1">Example Analysis</h1>



        

    </div>
</div>


        <div class="container">
    <div class="row">
        <main class="col-md-9 layout-page__main">
            <section class="mobile-toc border mb-3 p-3 d-md-none">
                <div class="d-flex align-items-center">
                    <h2 class="mb-0">Table of Contents</h2>

                    <button
                        class="js-only ml-2 toggle-toc"
                        aria-controls="mobileTOC"
                        aria-label="Toggle table of contents"
                        aria-expanded="true"
                    >
                        <span aria-hidden="true" data-role="toggle">Hide</span>
                    </button>
                </div>

                














<ul id="mobileTOC" class="js-toc mb-0 mt-2">
  <li><a href="#required-input-data">Required input data</a>
    <ul>
      <li><a href="#1-spatial-transcriptomics-data-eg">1. spatial transcriptomics data, e.g.,</a></li>
      <li><a href="#2-single-cell-rnaseq-scrna-seq-data--eg">2. single cell RNAseq ((scRNA-seq)) data,  e.g.,</a></li>
    </ul>
  </li>
  <li><a href="#cell-type-deconvolution">Cell Type Deconvolution</a>
    <ul>
      <li><a href="#1-create-an-card-object">1. Create an <code class="language-plaintext highlighter-rouge">CARD</code> object</a></li>
      <li><a href="#2-deconvolution-using-card">2. Deconvolution using CARD</a></li>
      <li><a href="#3-visualize-the-proportion-for-each-cell-type">3. Visualize the proportion for each cell type</a></li>
      <li><a href="#4-visualize-the-proportion-for-two-cell-types">4. Visualize the proportion for two cell types</a></li>
      <li><a href="#5-visualize-the-cell-type-proportion-correlation">5. Visualize the cell type proportion correlation</a></li>
    </ul>
  </li>
  <li><a href="#refined-spatial-map">Refined spatial map</a>
    <ul>
      <li><a href="#1-imputation-on-the-newly-grided-spatial-locations">1. Imputation on the newly grided spatial locations</a></li>
      <li><a href="#2-visualize-the-cell-type-proportion-at-an-enhanced-resolution">2. Visualize the cell type proportion at an enhanced resolution</a></li>
      <li><a href="#3-visualize-the-marker-gene-expression-at-an-enhanced-resolution">3. Visualize the marker gene expression at an enhanced resolution</a></li>
    </ul>
  </li>
  <li><a href="#extension-of-card-in-a-reference-free-version-cardfree">Extension of CARD in a reference-free version: CARDfree</a>
    <ul>
      <li><a href="#1-create-an-cardfree-object">1. Create an <code class="language-plaintext highlighter-rouge">CARDfree</code> object</a></li>
      <li><a href="#2-deconvolution-using-cardfree">2. Deconvolution using CARDfree</a></li>
      <li><a href="#3-visualization-of-the-results-of-cardfree">3. Visualization of the results of CARDfree</a></li>
    </ul>
  </li>
  <li><a href="#extension-of-card-for-single-cell-resolution-mapping">Extension of CARD for single cell resolution mapping</a></li>
</ul>


            </section>

            <section class="scope-markdown">
                













    <p>This tutorial is the example analysis with CARD on the human pancreatic ductal adenocarcinomas data from <a href="https://www.nature.com/articles/s41587-019-0392-8?proof=t">Moncada et al, 2020</a>. Before runing the tutorial, make sure that the CARD package is installed. Installation instructions see the <a href="https://kaizheng-academic.github.io/code/lrtm-2025-tcbb/documentation/02_installation.html">link</a></p>
      <h2 id="required-input-data">
        
        
          Required input data <a href="#required-input-data" class="." title="" >#</a>
        
        
      </h2>
<p><code class="language-plaintext highlighter-rouge">CARD</code> requires two types of input data:</p>
<ul>
  <li>spatial transcriptomics count data, along with spatial location information.</li>
  <li>single cell RNAseq (scRNA-seq) count data, along with meta information indicating the cell type information and the sample (subject) information for each cell.</li>
</ul>

<p>The example data for runing the tutorial can be downloaded in this <a href="https://kaizheng-academic.github.io/code/lrtm-2025-tcbb/documentation/03_data.html">page</a>
Here are the details about the required data input illustrated by the example datasets.</p>
    
      <h3 id="1-spatial-transcriptomics-data-eg">
        
        
          1. spatial transcriptomics data, e.g., <a href="#1-spatial-transcriptomics-data-eg" class="." title="" >#</a>
        
        
      </h3>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#### load the example spatial transcriptomics count data, </span><span class="w">
</span><span class="n">load</span><span class="p">(</span><span class="s2">"./spatial_count.RData"</span><span class="p">)</span><span class="w">
</span><span class="n">spatial_count</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span><span class="w">
</span><span class="m">4</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="n">sparse</span><span class="w"> </span><span class="n">Matrix</span><span class="w"> </span><span class="n">of</span><span class="w"> </span><span class="n">class</span><span class="w"> </span><span class="s2">"dgCMatrix"</span><span class="w">
           </span><span class="m">10</span><span class="n">x10</span><span class="w"> </span><span class="m">10</span><span class="n">x13</span><span class="w"> </span><span class="m">10</span><span class="n">x14</span><span class="w"> </span><span class="m">10</span><span class="n">x15</span><span class="w">
</span><span class="n">X5S_rRNA</span><span class="w">       </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">
</span><span class="n">X5_8S_rRNA</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">
</span><span class="n">X7SK</span><span class="w">           </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">
</span><span class="n">A1BG.AS1</span><span class="w">       </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">
</span></code></pre></div></div>
<p>The spatial transcriptomics count data must be in the format of matrix or sparseMatrix, while each row represents a gene and each column represents a spatial location. The column names of the spatial data can be in the “XcoordxYcoord” (i.e., 10x10) format, but you can also maintain your original spot names, for example, barcode names.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#### load the example spatial location data, </span><span class="w">
</span><span class="n">load</span><span class="p">(</span><span class="s2">"./spatial_location.RData"</span><span class="p">)</span><span class="w">
</span><span class="n">spatial_location</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,]</span><span class="w">
       </span><span class="n">x</span><span class="w">  </span><span class="n">y</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="m">10</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="m">13</span><span class="w">
</span><span class="m">10</span><span class="n">x14</span><span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="m">14</span><span class="w">
</span><span class="m">10</span><span class="n">x15</span><span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="m">15</span><span class="w">
</span></code></pre></div></div>
<p>The spatial location data must be in the format of data frame while each row represents a spatial location, the first column represents the x coordinate and the second column represents the y coordinate. The rownames of the spatial location data frame should match exactly with the column names of the spatial_count.</p>
    
      <h3 id="2-single-cell-rnaseq-scrna-seq-data--eg">
        
        
          2. single cell RNAseq ((scRNA-seq)) data,  e.g., <a href="#2-single-cell-rnaseq-scrna-seq-data--eg" class="." title="" >#</a>
        
        
      </h3>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#### load the example scRNA-seq count data, </span><span class="w">
</span><span class="n">load</span><span class="p">(</span><span class="s2">"./sc_count.RData"</span><span class="p">)</span><span class="w">
</span><span class="n">sc_count</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span><span class="w">
</span><span class="m">4</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="n">sparse</span><span class="w"> </span><span class="n">Matrix</span><span class="w"> </span><span class="n">of</span><span class="w"> </span><span class="n">class</span><span class="w"> </span><span class="s2">"dgCMatrix"</span><span class="w">
      </span><span class="n">Cell1</span><span class="w"> </span><span class="n">Cell2</span><span class="w"> </span><span class="n">Cell3</span><span class="w"> </span><span class="n">Cell4</span><span class="w">
</span><span class="n">A1BG</span><span class="w">      </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">
</span><span class="n">A1CF</span><span class="w">      </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="m">1</span><span class="w">
</span><span class="n">A2M</span><span class="w">       </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">
</span><span class="n">A2ML1</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">     </span><span class="n">.</span><span class="w">
</span></code></pre></div></div>
<p>The scRNA-seq count data must be in the format of matrix or sparseMatrix, while each row represents a gene and each column represents a cell.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#### load the example scRNA-seq meta data, </span><span class="w">
</span><span class="n">load</span><span class="p">(</span><span class="s2">"./sc_meta.RData"</span><span class="p">)</span><span class="w">
</span><span class="n">sc_meta</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,]</span><span class="w">
      </span><span class="n">cellID</span><span class="w">                             </span><span class="n">cellType</span><span class="w"> </span><span class="n">sampleInfo</span><span class="w">
</span><span class="n">Cell1</span><span class="w">  </span><span class="n">Cell1</span><span class="w">                         </span><span class="n">Acinar_cells</span><span class="w">    </span><span class="n">sample1</span><span class="w">
</span><span class="n">Cell2</span><span class="w">  </span><span class="n">Cell2</span><span class="w">          </span><span class="n">Ductal_terminal_ductal_like</span><span class="w">    </span><span class="n">sample1</span><span class="w">
</span><span class="n">Cell3</span><span class="w">  </span><span class="n">Cell3</span><span class="w">          </span><span class="n">Ductal_terminal_ductal_like</span><span class="w">    </span><span class="n">sample1</span><span class="w">
</span><span class="n">Cell4</span><span class="w">  </span><span class="n">Cell4</span><span class="w"> </span><span class="n">Ductal_CRISP3_high</span><span class="o">-</span><span class="n">centroacinar_like</span><span class="w">    </span><span class="n">sample1</span><span class="w">
</span></code></pre></div></div>
<p>The scRNAseq meta data must be in the format of data frame while each row represents a cell. The rownames of the scRNAseq meta data should match exactly with the column names of the scRNAseq count data. The sc_meta data must contain the column indicating the cell type assignment for each cell (e.g., “cellType” column in the example sc_meta data). Sample/subject information should be provided, if there is only one sample, we can add a column by <code class="language-plaintext highlighter-rouge">sc_meta$sampleInfo = "sample1"</code>.</p>

<p>We suggest the users to check their single cell RNASeq data carefully before running CARD. We suggest the users to input the single cell RNAseq data with each cell type containing at least 2 cells. i.e. print(table(sc_meta$cellType,useNA = “ifany”))</p>
    
      <h2 id="cell-type-deconvolution">
        
        
          Cell Type Deconvolution <a href="#cell-type-deconvolution" class="." title="" >#</a>
        
        
      </h2>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">CARD</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
    
      <h3 id="1-create-an-card-object">
        
        
          1. Create an <code class="language-plaintext highlighter-rouge">CARD</code> object <a href="#1-create-an-card-object" class="." title="" >#</a>
        
        
      </h3>
<p>The CARD object is created by the function <code class="language-plaintext highlighter-rouge">createCARDObject</code>. The essential inputs are:</p>
<ul>
  <li>sc_count: Matrix or sparse matrix of raw scRNA-seq count data, each row represents a gene and each column represents a cell. This sc_count data serves as a reference for the cell type deconvolution for spatial transcriptomics data.</li>
  <li>sc_meta: Data frame, with each row representing the cell type and/or sample information of a specific cell. The row names of this data frame should match exactly with the column names of the sc_count data. The sc_meta data must contain the column indicating the cell type assignment for each cell (e.g., “cellType” column in the example sc_meta data).</li>
  <li>spatial_count: Matrix or sparse matrix of raw spatial resolved transcriptomics count data, each row represents a gene and each column represents a spatial location. This is the spatial transcriptomics data that we are interested to deconvolute.</li>
  <li>spatial_location: Data frame, with two columns representing the x and y coordinates of the spatial location. The rownames of this data frame should match eaxctly with the columns of the spatial_count.</li>
  <li>ct.varname: Caracter, the name of the column in sc_meta that specifies the cell type assignment.</li>
  <li>ct.select: Vector of cell type names that you are interested in to deconvolute, default as NULL. If NULL, then use all cell types provided by single cell dataset.</li>
  <li>sample.varname: Character,the name of the column in sc_meta that specifies the sample/subject information. If NULL, we just use the whole data as one sample/subject.</li>
  <li>minCountGene: Numeric, include spatial locations where at least this number of counts detected. Default is 100.</li>
  <li>minCountSpot: Numeric, include genes where at least this number of spatial locations that have non-zero expression. Default is 5.</li>
</ul>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">CARD_obj</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">createCARDObject</span><span class="p">(</span><span class="w">
	</span><span class="n">sc_count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sc_count</span><span class="p">,</span><span class="w">
	</span><span class="n">sc_meta</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sc_meta</span><span class="p">,</span><span class="w">
	</span><span class="n">spatial_count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">spatial_count</span><span class="p">,</span><span class="w">
	</span><span class="n">spatial_location</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">spatial_location</span><span class="p">,</span><span class="w">
	</span><span class="n">ct.varname</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cellType"</span><span class="p">,</span><span class="w">
	</span><span class="n">ct.select</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unique</span><span class="p">(</span><span class="n">sc_meta</span><span class="o">$</span><span class="n">cellType</span><span class="p">),</span><span class="w">
	</span><span class="n">sample.varname</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"sampleInfo"</span><span class="p">,</span><span class="w">
	</span><span class="n">minCountGene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w">
	</span><span class="n">minCountSpot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> 
</span><span class="c1">## QC on scRNASeq dataset! ...</span><span class="w">
</span><span class="c1">## QC on spatially-resolved dataset! ..</span><span class="w">
</span></code></pre></div></div>
<p>The spatial data are stored in <code class="language-plaintext highlighter-rouge">CARD_obj@spatial_countMat</code> and <code class="language-plaintext highlighter-rouge">CARD_obj@spatial_location</code> while the scRNA-seq data is stored in CARD_obj@sc_eset in the format of SingleCellExperiment.</p>
    
      <h3 id="2-deconvolution-using-card">
        
        
          2. Deconvolution using CARD <a href="#2-deconvolution-using-card" class="." title="" >#</a>
        
        
      </h3>
<p>Now we have evrything stored in the CARD object, we can use CARD to deconvolute the spatial transcriptomics data. CARD is computationally fast and memory efficient. CARD relies on an efficient optimization algorithm for constrained maximum likelihood estimation and is scalable to spatial transcriptomics with tens of thousands of spatial locations and tens of thousands of genes. For the example dataset with the sample size of 428 locations, it takes within a minute to finish the deconvolution.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">CARD_obj</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_deconvolution</span><span class="p">(</span><span class="n">CARD_object</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_obj</span><span class="p">)</span><span class="w">
</span><span class="c1">## create reference matrix from scRNASeq...</span><span class="w">
</span><span class="c1">## Select Informative Genes! ...</span><span class="w">
</span><span class="c1">## Deconvolution Starts! ...</span><span class="w">
</span><span class="c1">## Deconvolution Finish! ...</span><span class="w">
</span></code></pre></div></div>
<p>The results are stored in <code class="language-plaintext highlighter-rouge">CARD_obj@Proportion_CARD</code>.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">print</span><span class="p">(</span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">Proportion_CARD</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,])</span><span class="w">
      </span><span class="n">Acinar_cells</span><span class="w"> </span><span class="n">Ductal_terminal_ductal_like</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w"> </span><span class="m">6.370860e-02</span><span class="w">                  </span><span class="m">0.02391251</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w"> </span><span class="m">7.818278e-08</span><span class="w">                  </span><span class="m">0.02996182</span><span class="w">
      </span><span class="n">Ductal_CRISP3_high</span><span class="o">-</span><span class="n">centroacinar_like</span><span class="w"> </span><span class="n">Cancer_clone_A</span><span class="w"> </span><span class="n">Ductal_MHC_Class_II</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w">                            </span><span class="m">0.1801753</span><span class="w">   </span><span class="m">4.229928e-04</span><span class="w">         </span><span class="m">0.021557706</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w">                            </span><span class="m">0.9620440</span><span class="w">   </span><span class="m">1.910394e-07</span><span class="w">         </span><span class="m">0.006437371</span><span class="w">
      </span><span class="n">Cancer_clone_B</span><span class="w">      </span><span class="n">mDCs_A</span><span class="w"> </span><span class="n">Ductal_APOL1_high</span><span class="o">-</span><span class="n">hypoxic</span><span class="w">   </span><span class="n">Tuft_cells</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w">   </span><span class="m">4.339480e-05</span><span class="w"> </span><span class="m">0.011136707</span><span class="w">              </span><span class="m">4.967235e-04</span><span class="w"> </span><span class="m">2.025089e-03</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w">   </span><span class="m">2.319262e-05</span><span class="w"> </span><span class="m">0.001013068</span><span class="w">              </span><span class="m">3.864917e-06</span><span class="w"> </span><span class="m">1.796433e-06</span><span class="w">
            </span><span class="n">mDCs_B</span><span class="w">         </span><span class="n">pDCs</span><span class="w"> </span><span class="n">Endocrine_cells</span><span class="w"> </span><span class="n">Endothelial_cells</span><span class="w"> </span><span class="n">Macrophages_A</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w"> </span><span class="m">0.0792525811</span><span class="w"> </span><span class="m">7.432979e-07</span><span class="w">    </span><span class="m">6.848627e-03</span><span class="w">      </span><span class="m">1.722855e-01</span><span class="w">  </span><span class="m">9.909662e-02</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w"> </span><span class="m">0.0004695018</span><span class="w"> </span><span class="m">1.566377e-11</span><span class="w">    </span><span class="m">3.925412e-11</span><span class="w">      </span><span class="m">4.198468e-11</span><span class="w">  </span><span class="m">2.766705e-05</span><span class="w">
        </span><span class="n">Mast_cells</span><span class="w"> </span><span class="n">Macrophages_B</span><span class="w"> </span><span class="n">T_cells_</span><span class="o">&amp;</span><span class="err">_</span><span class="n">NK_cells</span><span class="w">    </span><span class="n">Monocytes</span><span class="w">        </span><span class="n">RBCs</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w"> </span><span class="m">7.411147e-11</span><span class="w">  </span><span class="m">3.090675e-02</span><span class="w">       </span><span class="m">4.976256e-06</span><span class="w"> </span><span class="m">2.663846e-06</span><span class="w"> </span><span class="m">3.84187e-10</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w"> </span><span class="m">2.387132e-11</span><span class="w">  </span><span class="m">9.499908e-06</span><span class="w">       </span><span class="m">1.172387e-11</span><span class="w"> </span><span class="m">2.000116e-06</span><span class="w"> </span><span class="m">1.00967e-06</span><span class="w">
       </span><span class="n">Fibroblasts</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w"> </span><span class="m">3.081225e-01</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w"> </span><span class="m">4.898874e-06</span><span class="w">
</span></code></pre></div></div>
    
      <h3 id="3-visualize-the-proportion-for-each-cell-type">
        
        
          3. Visualize the proportion for each cell type <a href="#3-visualize-the-proportion-for-each-cell-type" class="." title="" >#</a>
        
        
      </h3>
<p>First, we jointly visualize the cell type proportion matrix through scatterpie plot. Note that here because the number of spots is relatively small, so jointly visualize the cell type proportion matrix in the scatterpie plot format is duable. We do not recommend users to visualize this plot when the number of spots is &gt; 500. Instead, we recommend users to visualize the proportion directly, i.e., using the function CARD.visualize.prop(). Details of using this function see the next example.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## set the colors. Here, I just use the colors in the manuscript, if the color is not provided, the function will use default color in the package. </span><span class="w">
</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"#FFD92F"</span><span class="p">,</span><span class="s2">"#4DAF4A"</span><span class="p">,</span><span class="s2">"#FCCDE5"</span><span class="p">,</span><span class="s2">"#D9D9D9"</span><span class="p">,</span><span class="s2">"#377EB8"</span><span class="p">,</span><span class="s2">"#7FC97F"</span><span class="p">,</span><span class="s2">"#BEAED4"</span><span class="p">,</span><span class="w">
    </span><span class="s2">"#FDC086"</span><span class="p">,</span><span class="s2">"#FFFF99"</span><span class="p">,</span><span class="s2">"#386CB0"</span><span class="p">,</span><span class="s2">"#F0027F"</span><span class="p">,</span><span class="s2">"#BF5B17"</span><span class="p">,</span><span class="s2">"#666666"</span><span class="p">,</span><span class="s2">"#1B9E77"</span><span class="p">,</span><span class="s2">"#D95F02"</span><span class="p">,</span><span class="w">
    </span><span class="s2">"#7570B3"</span><span class="p">,</span><span class="s2">"#E7298A"</span><span class="p">,</span><span class="s2">"#66A61E"</span><span class="p">,</span><span class="s2">"#E6AB02"</span><span class="p">,</span><span class="s2">"#A6761D"</span><span class="p">)</span><span class="w">
</span><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">CARD.visualize.pie</span><span class="p">(</span><span class="w">
	</span><span class="n">proportion</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">Proportion_CARD</span><span class="p">,</span><span class="w">
	</span><span class="n">spatial_location</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">spatial_location</span><span class="p">,</span><span class="w"> 
 	</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colors</span><span class="p">,</span><span class="w"> 
  	</span><span class="n">radius</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.52</span><span class="p">)</span><span class="w"> </span><span class="c1">### You can choose radius = NULL or your own radius number</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p1</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>Here is an example output: 
<img src="/code/lrtm-2025-tcbb/documentation/Example_analysis_visualizePie.png" alt="Example_Pie" /></p>

<p>Then, we can select some interested cell types to visualize separately.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## select the cell type that we are interested</span><span class="w">
</span><span class="n">ct.visualize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Acinar_cells"</span><span class="p">,</span><span class="s2">"Cancer_clone_A"</span><span class="p">,</span><span class="s2">"Cancer_clone_B"</span><span class="p">,</span><span class="s2">"Ductal_terminal_ductal_like"</span><span class="p">,</span><span class="s2">"Ductal_CRISP3_high-centroacinar_like"</span><span class="p">,</span><span class="s2">"Ductal_MHC_Class_II"</span><span class="p">,</span><span class="s2">"Ductal_APOL1_high-hypoxic"</span><span class="p">,</span><span class="s2">"Fibroblasts"</span><span class="p">)</span><span class="w">
</span><span class="c1">## visualize the spatial distribution of the cell type proportion</span><span class="w">
</span><span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">CARD.visualize.prop</span><span class="p">(</span><span class="w">
	</span><span class="n">proportion</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">Proportion_CARD</span><span class="p">,</span><span class="w">        
	</span><span class="n">spatial_location</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">spatial_location</span><span class="p">,</span><span class="w"> 
	</span><span class="n">ct.visualize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ct.visualize</span><span class="p">,</span><span class="w">                 </span><span class="c1">### selected cell types to visualize</span><span class="w">
	</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"lightblue"</span><span class="p">,</span><span class="s2">"lightyellow"</span><span class="p">,</span><span class="s2">"red"</span><span class="p">),</span><span class="w"> </span><span class="c1">### if not provide, we will use the default colors</span><span class="w">
	</span><span class="n">NumCols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="p">,</span><span class="w">                                 </span><span class="c1">### number of columns in the figure panel</span><span class="w">
        </span><span class="n">pointSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3.0</span><span class="p">)</span><span class="w">                             </span><span class="c1">### point size in ggplot2 scatterplot  </span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p2</span><span class="p">)</span><span class="w">

</span></code></pre></div></div>
<p>Here is an example output: 
<img src="/code/lrtm-2025-tcbb/documentation/Example_analysis_visualizeProp.png" alt="Example_Prop" /></p>
    
      <h3 id="4-visualize-the-proportion-for-two-cell-types">
        
        
          4. Visualize the proportion for two cell types <a href="#4-visualize-the-proportion-for-two-cell-types" class="." title="" >#</a>
        
        
      </h3>
<p>Motivated by this github <a href="https://github.com/YingMa0107/CARD/issues/25#issuecomment-1550231037">issue post</a>, we added a new visualization function to visualize the distribution of two cell types on the same post.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## visualize the spatial distribution of two cell types on the same plot
p3 = CARD.visualize.prop.2CT(
proportion = CARD_obj@Proportion_CARD,                             ### Cell type proportion estimated by CARD
spatial_location = CARD_obj@spatial_location,                      ### spatial location information
ct2.visualize = c("Cancer_clone_A","Cancer_clone_B"),              ### two cell types you want to visualize
colors = list(c("lightblue","lightyellow","red"),c("lightblue","lightyellow","black")))       ### two color scales                             
print(p3)

</code></pre></div></div>
<p>Here is an example output:</p>
<p align="center"> 
<img src="Example_analysis_visualizeProp_2CT.png" width="500" />
</p>
    
      <h3 id="5-visualize-the-cell-type-proportion-correlation">
        
        
          5. Visualize the cell type proportion correlation <a href="#5-visualize-the-cell-type-proportion-correlation" class="." title="" >#</a>
        
        
      </h3>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">p4</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">CARD.visualize.Cor</span><span class="p">(</span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">Proportion_CARD</span><span class="p">,</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">)</span><span class="w"> </span><span class="c1"># if not provide, we will use the default colors</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p4</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>Here is an example output:</p>
<p align="left"> 
<img src="Example_analysis_visualizeCor.png" width="700" />
</p>
    
      <h2 id="refined-spatial-map">
        
        
          Refined spatial map <a href="#refined-spatial-map" class="." title="" >#</a>
        
        
      </h2>
<p>A unique feature of CARD is its ability to model the spatial correlation in cell type composition across tissue locations, thus enabling spatially informed cell type deconvolution. Modeling spatial correlation allows us to not only accurately infer the cell type composition on each spatial location, but also impute cell type compositions and gene expression levels on unmeasured tissue locations, facilitating the construction of a refined spatial tissue map with a resolution much higher than that measured in the original study.
Specifically, CARD constructed a refined spatial map through the function <code class="language-plaintext highlighter-rouge">CARD.imputation</code>. The essential inputs are:</p>

<ul>
  <li>CARD_object: CARD Object with estimated cell type compositions on the original spatial resolved transcriptomics data.</li>
  <li>NumGrids: Initial number of newly grided spatial locations. The final number of newly grided spatial locations will be lower than this value since the newly grided locations outside the shape of the tissue will be filtered.</li>
  <li>ineibor: Numeric, number of neighbors used in the imputation on newly grided spatial locations, default is 10.</li>
</ul>

<p>Briefly, CARD first outlined the shape of the tissue by applying a two-dimensional concave hull algorithm on the existing locations, then perform imputation on the newly grided spatial locations. We recommend to check the exisiting spatial locations to see if there are outliers that are seriously affect the detection of the shape.</p>
    
      <h3 id="1-imputation-on-the-newly-grided-spatial-locations">
        
        
          1. Imputation on the newly grided spatial locations <a href="#1-imputation-on-the-newly-grided-spatial-locations" class="." title="" >#</a>
        
        
      </h3>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">CARD_obj</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD.imputation</span><span class="p">(</span><span class="n">CARD_obj</span><span class="p">,</span><span class="n">NumGrids</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2000</span><span class="p">,</span><span class="n">ineibor</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="n">exclude</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">)</span><span class="w">
</span><span class="c1">## The rownames of locations are matched ...</span><span class="w">
</span><span class="c1">## Make grids on new spatial locations ...</span><span class="w">
</span></code></pre></div></div>
<p>The results are store in <code class="language-plaintext highlighter-rouge">CARD_obj@refined_prop</code> and <code class="language-plaintext highlighter-rouge">CARD_obj@refined_expression</code></p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## Visualize the newly grided spatial locations to see if the shape is correctly detected. If not, the user can provide the row names of the excluded spatial location data into the CARD.imputation function</span><span class="w">
</span><span class="n">location_imputation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cbind.data.frame</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">sapply</span><span class="p">(</span><span class="n">strsplit</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">refined_prop</span><span class="p">),</span><span class="n">split</span><span class="o">=</span><span class="s2">"x"</span><span class="p">),</span><span class="s2">"["</span><span class="p">,</span><span class="m">1</span><span class="p">)),</span><span class="w">
	</span><span class="n">y</span><span class="o">=</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">sapply</span><span class="p">(</span><span class="n">strsplit</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">refined_prop</span><span class="p">),</span><span class="n">split</span><span class="o">=</span><span class="s2">"x"</span><span class="p">),</span><span class="s2">"["</span><span class="p">,</span><span class="m">2</span><span class="p">)))</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">location_imputation</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">refined_prop</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">
</span><span class="n">p5</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">location_imputation</span><span class="p">,</span><span class="w"> 
       </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">geom_point</span><span class="p">(</span><span class="n">shape</span><span class="o">=</span><span class="m">22</span><span class="p">,</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#7dc7f5"</span><span class="p">)</span><span class="o">+</span><span class="w">
</span><span class="n">theme</span><span class="p">(</span><span class="n">plot.margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">margin</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span><span class="w"> </span><span class="m">0.1</span><span class="p">,</span><span class="w"> </span><span class="m">0.1</span><span class="p">,</span><span class="w"> </span><span class="m">0.1</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">
    </span><span class="n">legend.position</span><span class="o">=</span><span class="s2">"bottom"</span><span class="p">,</span><span class="w">
    </span><span class="n">panel.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
    </span><span class="n">plot.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
    </span><span class="n">panel.border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey89"</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p5</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>Here is the example of the newly grided spatial locations. The shape outline is detected well.</p>
<p align="center"> 
<img src="Example_analysis_grids.png" width="400" />
</p>
    
      <h3 id="2-visualize-the-cell-type-proportion-at-an-enhanced-resolution">
        
        
          2. Visualize the cell type proportion at an enhanced resolution <a href="#2-visualize-the-cell-type-proportion-at-an-enhanced-resolution" class="." title="" >#</a>
        
        
      </h3>
<p>Now we can use the same <code class="language-plaintext highlighter-rouge">CARD.visualize.prop</code> function to visualize the cell type proportion at the enhanced resolution. But this time, the input of the function should be the imputed cell typr propotion and corresponding newly grided spatial locations.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">p6</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">CARD.visualize.prop</span><span class="p">(</span><span class="w">
	</span><span class="n">proportion</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">refined_prop</span><span class="p">,</span><span class="w">                         
	</span><span class="n">spatial_location</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">location_imputation</span><span class="p">,</span><span class="w">            
	</span><span class="n">ct.visualize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ct.visualize</span><span class="p">,</span><span class="w">                    
	</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"lightblue"</span><span class="p">,</span><span class="s2">"lightyellow"</span><span class="p">,</span><span class="s2">"red"</span><span class="p">),</span><span class="w">    
	</span><span class="n">NumCols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="p">)</span><span class="w">                                  
</span><span class="n">print</span><span class="p">(</span><span class="n">p6</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><img src="/code/lrtm-2025-tcbb/documentation/Example_analysis_grid_prop.png" alt="Example_grids" /></p>
    
      <h3 id="3-visualize-the-marker-gene-expression-at-an-enhanced-resolution">
        
        
          3. Visualize the marker gene expression at an enhanced resolution <a href="#3-visualize-the-marker-gene-expression-at-an-enhanced-resolution" class="." title="" >#</a>
        
        
      </h3>
<p>After we obtained cell type proportion at the enhanced resolution by CARD, we can predict the spatial gene expression at the enhanced resolution. The following code is to visualize the marker gene expression at an enhanced resolution.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">p7</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">CARD.visualize.gene</span><span class="p">(</span><span class="w">
	</span><span class="n">spatial_expression</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">refined_expression</span><span class="p">,</span><span class="w">
	</span><span class="n">spatial_location</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">location_imputation</span><span class="p">,</span><span class="w">
	</span><span class="n">gene.visualize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Tm4sf1"</span><span class="p">,</span><span class="s2">"S100a4"</span><span class="p">,</span><span class="s2">"Tff3"</span><span class="p">,</span><span class="s2">"Apol1"</span><span class="p">,</span><span class="s2">"Crisp3"</span><span class="p">,</span><span class="s2">"CD248"</span><span class="p">),</span><span class="w">
	</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w">
	</span><span class="n">NumCols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p7</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><img src="/code/lrtm-2025-tcbb/documentation/Example_analysis_grid_Gene.png" alt="Example_grid_Gene" /></p>

<p>Now, compare with the original resolution, CARD facilitates the construction of a refined spatial tissue map with a resolution much higher than that measured in the original study.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">p8</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">CARD.visualize.gene</span><span class="p">(</span><span class="w">
	</span><span class="n">spatial_expression</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">spatial_countMat</span><span class="p">,</span><span class="w">
	</span><span class="n">spatial_location</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_obj</span><span class="o">@</span><span class="n">spatial_location</span><span class="p">,</span><span class="w">
	</span><span class="n">gene.visualize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Tm4sf1"</span><span class="p">,</span><span class="s2">"S100a4"</span><span class="p">,</span><span class="s2">"Tff3"</span><span class="p">,</span><span class="s2">"Apol1"</span><span class="p">,</span><span class="s2">"Crisp3"</span><span class="p">,</span><span class="s2">"CD248"</span><span class="p">),</span><span class="w">
	</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w">
	</span><span class="n">NumCols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p8</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><img src="/code/lrtm-2025-tcbb/documentation/Example_analysis_Original_Gene.png" alt="Example_grid_Gene" /></p>
    
      <h2 id="extension-of-card-in-a-reference-free-version-cardfree">
        
        
          Extension of CARD in a reference-free version: CARDfree <a href="#extension-of-card-in-a-reference-free-version-cardfree" class="." title="" >#</a>
        
        
      </h2>
<p>We extended CARD to enable reference-free cell type deconvolution and eliminate the need for the single-cell reference data. We refer to this extension as the reference-free version of CARD, or simply as CARDfree. Different from CARD, CARDfree no longer requires an external single-cell reference dataset and only needs users to input a list of gene names for previously known cell type markers. We use the same exmple dataset to illustrate the use of CARDfree. In addition to the exmple dataset, CARDfree also requires the input of marker gene list, which is in a list format with each element of the list being the cell type specific gene markers. The example marker list for runing the tutorial can be downloaded in this <a href="https://kaizheng-academic.github.io/code/lrtm-2025-tcbb/documentation/03_data.html">page</a>.</p>

<p>Similar to CARD, we will first need to create a CARDfree object with the spatial transcriptomics dataset and the marker gene list</p>
    
      <h3 id="1-create-an-cardfree-object">
        
        
          1. Create an <code class="language-plaintext highlighter-rouge">CARDfree</code> object <a href="#1-create-an-cardfree-object" class="." title="" >#</a>
        
        
      </h3>
<p>The CARDfree object is created by the function <code class="language-plaintext highlighter-rouge">createCARDfreeObject</code>. Briefly, the essential inputs are the same as the function <code class="language-plaintext highlighter-rouge">createCARDObject</code>, except that this function does not require the single cell count and meta information matrix. Instead, it requires a markerList.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## load the marker gene list</span><span class="w">
</span><span class="n">load</span><span class="p">(</span><span class="s2">"./markerList.RData"</span><span class="p">)</span><span class="w">
</span><span class="n">CARDfree_obj</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">createCARDfreeObject</span><span class="p">(</span><span class="w">
	</span><span class="n">markerList</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">markerList</span><span class="p">,</span><span class="w">
	</span><span class="n">spatial_count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">spatial_count</span><span class="p">,</span><span class="w">
	</span><span class="n">spatial_location</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">spatial_location</span><span class="p">,</span><span class="w">
	</span><span class="n">minCountGene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w">
	</span><span class="n">minCountSpot</span><span class="w"> </span><span class="o">=</span><span class="m">5</span><span class="p">)</span><span class="w"> 
</span></code></pre></div></div>
<p>The spatial data are stored in <code class="language-plaintext highlighter-rouge">CARDfree_obj@spatial_countMat</code> and <code class="language-plaintext highlighter-rouge">CARDfree_obj@spatial_location</code> while the marker list is stored in CARDfree_objj@markerList in the format of list.</p>
    
      <h3 id="2-deconvolution-using-cardfree">
        
        
          2. Deconvolution using CARDfree <a href="#2-deconvolution-using-cardfree" class="." title="" >#</a>
        
        
      </h3>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## deconvolution using CARDfree</span><span class="w">
</span><span class="n">CARDfree_obj</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_refFree</span><span class="p">(</span><span class="n">CARDfree_obj</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>The results are stored in <code class="language-plaintext highlighter-rouge">CARDfree_obj@Proportion_CARD</code>.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## One limitation of reference-free version of CARD is that the cell types inferred from CARDfree do not come with a cell type label. It might be difficult to interpret the results. </span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">CARDfree_obj</span><span class="o">@</span><span class="n">Proportion_CARD</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,])</span><span class="w">
            </span><span class="n">CT1</span><span class="w">          </span><span class="n">CT2</span><span class="w">          </span><span class="n">CT3</span><span class="w">          </span><span class="n">CT4</span><span class="w">          </span><span class="n">CT5</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w"> </span><span class="m">0.1293363</span><span class="w"> </span><span class="m">8.155678e-12</span><span class="w"> </span><span class="m">1.342871e-10</span><span class="w"> </span><span class="m">7.934516e-06</span><span class="w"> </span><span class="m">1.029232e-01</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w"> </span><span class="m">0.7580073</span><span class="w"> </span><span class="m">1.241726e-23</span><span class="w"> </span><span class="m">1.981989e-27</span><span class="w"> </span><span class="m">3.367648e-40</span><span class="w"> </span><span class="m">1.577430e-37</span><span class="w">
               </span><span class="n">CT6</span><span class="w">          </span><span class="n">CT7</span><span class="w">          </span><span class="n">CT8</span><span class="w">          </span><span class="n">CT9</span><span class="w">         </span><span class="n">CT10</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w"> </span><span class="m">6.081635e-02</span><span class="w"> </span><span class="m">3.932038e-04</span><span class="w"> </span><span class="m">4.996482e-07</span><span class="w"> </span><span class="m">1.862520e-08</span><span class="w"> </span><span class="m">4.092203e-12</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w"> </span><span class="m">2.162971e-27</span><span class="w"> </span><span class="m">1.934013e-25</span><span class="w"> </span><span class="m">3.096247e-87</span><span class="w"> </span><span class="m">1.903368e-08</span><span class="w"> </span><span class="m">2.520620e-02</span><span class="w">
            </span><span class="n">CT11</span><span class="w">       </span><span class="n">CT12</span><span class="w">       </span><span class="n">CT13</span><span class="w">        </span><span class="n">CT14</span><span class="w">         </span><span class="n">CT15</span><span class="w">         </span><span class="n">CT16</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w"> </span><span class="m">0.05317422</span><span class="w"> </span><span class="m">0.22235226</span><span class="w"> </span><span class="m">0.05971717</span><span class="w"> </span><span class="m">0.174335930</span><span class="w"> </span><span class="m">5.853567e-38</span><span class="w"> </span><span class="m">2.537400e-10</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w"> </span><span class="m">0.02868479</span><span class="w"> </span><span class="m">0.08371092</span><span class="w"> </span><span class="m">0.09967332</span><span class="w"> </span><span class="m">0.004615396</span><span class="w"> </span><span class="m">4.214875e-37</span><span class="w"> </span><span class="m">1.017432e-04</span><span class="w">
              </span><span class="n">CT17</span><span class="w">         </span><span class="n">CT18</span><span class="w">         </span><span class="n">CT19</span><span class="w">         </span><span class="n">CT20</span><span class="w">
</span><span class="m">10</span><span class="n">x10</span><span class="w"> </span><span class="m">8.433169e-11</span><span class="w"> </span><span class="m">1.219174e-01</span><span class="w"> </span><span class="m">1.077052e-16</span><span class="w"> </span><span class="m">7.502541e-02</span><span class="w">
</span><span class="m">10</span><span class="n">x13</span><span class="w"> </span><span class="m">3.462751e-07</span><span class="w"> </span><span class="m">7.941545e-64</span><span class="w"> </span><span class="m">9.548596e-20</span><span class="w"> </span><span class="m">3.760342e-13</span><span class="w">
</span></code></pre></div></div>
    
      <h3 id="3-visualization-of-the-results-of-cardfree">
        
        
          3. Visualization of the results of CARDfree <a href="#3-visualization-of-the-results-of-cardfree" class="." title="" >#</a>
        
        
      </h3>
<p>Note that here because the number of spots is relatively small, so jointly visualize the cell type proportion matrix in the scatterpie plot format is duable. We do not recommend users to visualize this plot when the number of spots is &gt; 500. Instead, we recommend users to visualize the proportion directly, i.e., using the function CARD.visualize.prop().</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"#FFD92F"</span><span class="p">,</span><span class="s2">"#4DAF4A"</span><span class="p">,</span><span class="s2">"#FCCDE5"</span><span class="p">,</span><span class="s2">"#D9D9D9"</span><span class="p">,</span><span class="s2">"#377EB8"</span><span class="p">,</span><span class="s2">"#7FC97F"</span><span class="p">,</span><span class="s2">"#BEAED4"</span><span class="p">,</span><span class="s2">"#FDC086"</span><span class="p">,</span><span class="s2">"#FFFF99"</span><span class="p">,</span><span class="s2">"#386CB0"</span><span class="p">,</span><span class="s2">"#F0027F"</span><span class="p">,</span><span class="s2">"#BF5B17"</span><span class="p">,</span><span class="s2">"#666666"</span><span class="p">,</span><span class="s2">"#1B9E77"</span><span class="p">,</span><span class="s2">"#D95F02"</span><span class="p">,</span><span class="s2">"#7570B3"</span><span class="p">,</span><span class="s2">"#E7298A"</span><span class="p">,</span><span class="s2">"#66A61E"</span><span class="p">,</span><span class="s2">"#E6AB02"</span><span class="p">,</span><span class="s2">"#A6761D"</span><span class="p">)</span><span class="w">
</span><span class="c1">### In order to maximumply match with the original results of CARD, we order the colors to generally match with the results infered by CARD</span><span class="w">
</span><span class="n">CARDfree_obj</span><span class="o">@</span><span class="n">Proportion_CARD</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARDfree_obj</span><span class="o">@</span><span class="n">Proportion_CARD</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="m">8</span><span class="p">,</span><span class="m">10</span><span class="p">,</span><span class="m">14</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">6</span><span class="p">,</span><span class="m">12</span><span class="p">,</span><span class="m">18</span><span class="p">,</span><span class="m">7</span><span class="p">,</span><span class="m">13</span><span class="p">,</span><span class="m">20</span><span class="p">,</span><span class="m">19</span><span class="p">,</span><span class="m">16</span><span class="p">,</span><span class="m">17</span><span class="p">,</span><span class="m">11</span><span class="p">,</span><span class="m">15</span><span class="p">,</span><span class="m">4</span><span class="p">,</span><span class="m">9</span><span class="p">,</span><span class="m">3</span><span class="p">,</span><span class="m">5</span><span class="p">)]</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">CARDfree_obj</span><span class="o">@</span><span class="n">Proportion_CARD</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"CT"</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">)</span><span class="w">
</span><span class="n">p9</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">CARD.visualize.pie</span><span class="p">(</span><span class="n">CARDfree_obj</span><span class="o">@</span><span class="n">Proportion_CARD</span><span class="p">,</span><span class="n">CARDfree_obj</span><span class="o">@</span><span class="n">spatial_location</span><span class="p">,</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colors</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p9</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><img src="/code/lrtm-2025-tcbb/documentation/Example_analysis_CARDfree_prop.png" alt="Example_CARDfree" /></p>
    
      <h2 id="extension-of-card-for-single-cell-resolution-mapping">
        
        
          Extension of CARD for single cell resolution mapping <a href="#extension-of-card-for-single-cell-resolution-mapping" class="." title="" >#</a>
        
        
      </h2>
<p>We also extended CARD to facilitate the construction of single-cell resolution spatial transcriptomics from non-single-cell resolution spatial transcriptomics. Details of the algorithm see the main text. Briefly, we infer the single cell resolution gene expression for each measured spatial location from the non-single cell resolution spatial transcriptomics data based on reference scRNaseq data we used for deconvolution. The procedure is implemented in the function <code class="language-plaintext highlighter-rouge">CARD_SCMapping</code>. The essential inputs are:</p>
<ul>
  <li>CARD_object: CARD object create by the createCARDObject function. This one should be the one after we finish the deconvolution procedure</li>
  <li>shapeSpot: a character indicating whether the sampled spatial coordinates for single cells locating in a Square-like region or a Circle-like region. The center of this region is the measured spatial location in the non-single cell resolution spatial transcriptomics data. The default is “Square”, and the other option is “Circle”</li>
  <li>numCell: a numeric value indicating the number of cores used to accelerate the procedure.
    <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#### Note that here the shapeSpot is the user defined variable which indicates the capturing area of single cells. Details see above.</span><span class="w">
</span><span class="n">scMapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CARD_SCMapping</span><span class="p">(</span><span class="n">CARD_obj</span><span class="p">,</span><span class="n">shapeSpot</span><span class="o">=</span><span class="s2">"Square"</span><span class="p">,</span><span class="n">numCell</span><span class="o">=</span><span class="m">20</span><span class="p">,</span><span class="n">ncore</span><span class="o">=</span><span class="m">10</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">scMapping</span><span class="p">)</span><span class="w">
</span><span class="c1">### the output</span><span class="w">
</span><span class="n">class</span><span class="o">:</span><span class="w"> </span><span class="n">SingleCellExperiment</span><span class="w"> 
</span><span class="n">dim</span><span class="o">:</span><span class="w"> </span><span class="m">16381</span><span class="w"> </span><span class="m">8560</span><span class="w"> 
</span><span class="n">metadata</span><span class="p">(</span><span class="m">0</span><span class="p">)</span><span class="o">:</span><span class="w">
</span><span class="n">assays</span><span class="p">(</span><span class="m">1</span><span class="p">)</span><span class="o">:</span><span class="w"> </span><span class="n">counts</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="m">16381</span><span class="p">)</span><span class="o">:</span><span class="w"> </span><span class="n">A1BG</span><span class="w"> </span><span class="n">A1CF</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">ZZEF1</span><span class="w"> </span><span class="n">ZZZ3</span><span class="w">
</span><span class="n">rowData</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="m">1</span><span class="p">)</span><span class="o">:</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">count_CT</span><span class="p">)</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="m">8560</span><span class="p">)</span><span class="o">:</span><span class="w"> </span><span class="n">Cell686</span><span class="o">:</span><span class="m">10</span><span class="n">x10</span><span class="o">:</span><span class="m">9.97518192091957</span><span class="n">x9.83765210071579</span><span class="w">
</span><span class="n">Cell734</span><span class="o">:</span><span class="m">10</span><span class="n">x10</span><span class="o">:</span><span class="m">10.1896061601583</span><span class="n">x9.94081321195699</span><span class="w"> </span><span class="n">...</span><span class="w">
</span><span class="n">Cell443</span><span class="o">:</span><span class="m">9</span><span class="n">x33</span><span class="o">:</span><span class="m">9.4747460691724</span><span class="n">x32.5644472888671</span><span class="w">
</span><span class="n">Cell1488</span><span class="o">:</span><span class="m">9</span><span class="n">x33</span><span class="o">:</span><span class="m">9.43348842463456</span><span class="n">x33.4998327996582</span><span class="w">
</span><span class="n">colData</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="m">7</span><span class="p">)</span><span class="o">:</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">CT</span><span class="w"> </span><span class="n">Cell</span><span class="w">
</span><span class="n">reducedDimNames</span><span class="p">(</span><span class="m">0</span><span class="p">)</span><span class="o">:</span><span class="w">
</span><span class="n">mainExpName</span><span class="o">:</span><span class="w"> </span><span class="kc">NULL</span><span class="w">
</span><span class="n">altExpNames</span><span class="p">(</span><span class="m">0</span><span class="p">)</span><span class="o">:</span><span class="w">
</span><span class="c1">### spatial location info and expression count of the single cell resolution data</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">SingleCellExperiment</span><span class="p">)</span><span class="w">
</span><span class="n">MapCellCords</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">colData</span><span class="p">(</span><span class="n">scMapping</span><span class="p">))</span><span class="w">
</span><span class="n">count_SC</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">assays</span><span class="p">(</span><span class="n">scMapping</span><span class="p">)</span><span class="o">$</span><span class="n">counts</span><span class="w">
</span></code></pre></div>    </div>
    <p>The results are stored in a SingleCellExperiment object with mapped single cell resolution counts stored in the assays slot and the information of the spatial location for each single cell as well as their relashionship to the original measured spatial location is stored in the colData slot.</p>
  </li>
</ul>

<p>Next, we visualize the cell type for each single cell with their spatial location information</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">
</span><span class="n">df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">MapCellCords</span><span class="w">
</span><span class="n">colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"#8DD3C7"</span><span class="p">,</span><span class="s2">"#CFECBB"</span><span class="p">,</span><span class="s2">"#F4F4B9"</span><span class="p">,</span><span class="s2">"#CFCCCF"</span><span class="p">,</span><span class="s2">"#D1A7B9"</span><span class="p">,</span><span class="s2">"#E9D3DE"</span><span class="p">,</span><span class="s2">"#F4867C"</span><span class="p">,</span><span class="s2">"#C0979F"</span><span class="p">,</span><span class="w">
	</span><span class="s2">"#D5CFD6"</span><span class="p">,</span><span class="s2">"#86B1CD"</span><span class="p">,</span><span class="s2">"#CEB28B"</span><span class="p">,</span><span class="s2">"#EDBC63"</span><span class="p">,</span><span class="s2">"#C59CC5"</span><span class="p">,</span><span class="s2">"#C09CBF"</span><span class="p">,</span><span class="s2">"#C2D567"</span><span class="p">,</span><span class="s2">"#C9DAC3"</span><span class="p">,</span><span class="s2">"#E1EBA0"</span><span class="p">,</span><span class="w">
	</span><span class="s2">"#FFED6F"</span><span class="p">,</span><span class="s2">"#CDD796"</span><span class="p">,</span><span class="s2">"#F8CDDE"</span><span class="p">)</span><span class="w">
</span><span class="n">p10</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">CT</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
    </span><span class="n">geom_point</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3.0</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">scale_colour_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w">  </span><span class="n">colors</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="c1">#facet_wrap(~Method,ncol = 2,nrow = 3) + </span><span class="w">
        </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">margin</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span><span class="w"> </span><span class="m">0.1</span><span class="p">,</span><span class="w"> </span><span class="m">0.1</span><span class="p">,</span><span class="w"> </span><span class="m">0.1</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">
              </span><span class="n">panel.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="s2">"white"</span><span class="p">),</span><span class="w">
              </span><span class="n">plot.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="s2">"white"</span><span class="p">),</span><span class="w">
    </span><span class="n">legend.position</span><span class="o">=</span><span class="s2">"bottom"</span><span class="p">,</span><span class="w">
    </span><span class="n">panel.border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey89"</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">0.5</span><span class="p">),</span><span class="w">
    </span><span class="n">axis.text</span><span class="w"> </span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
    </span><span class="n">axis.ticks</span><span class="w"> </span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
    </span><span class="n">axis.title</span><span class="w"> </span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
    </span><span class="n">legend.title</span><span class="o">=</span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">13</span><span class="p">,</span><span class="n">face</span><span class="o">=</span><span class="s2">"bold"</span><span class="p">),</span><span class="w">
    </span><span class="n">legend.text</span><span class="o">=</span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">),</span><span class="w">
    </span><span class="n">legend.key</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"transparent"</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"white"</span><span class="p">),</span><span class="w">
    </span><span class="n">legend.key.size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.45</span><span class="p">,</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">),</span><span class="w">
    </span><span class="n">strip.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="n">face</span><span class="o">=</span><span class="s2">"bold"</span><span class="p">))</span><span class="o">+</span><span class="w">
                                </span><span class="n">guides</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="n">guide_legend</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">"Cell Type"</span><span class="p">))</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">p10</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><img src="/code/lrtm-2025-tcbb/documentation/Example_analysis_scMapping.png" alt="Example_grids" /></p>



            </section>
        </main>

        <aside class="col-md-3 layout-page__sidebar d-none d-md-block hidden-print">
            
            <nav id="page-toc" class="page-sidebar">
    














<ul class="list-unstyled pl-0">
  <li><a href="#required-input-data" class="nav-link px-3 py-1">Required input data</a>
    <ul>
      <li><a href="#1-spatial-transcriptomics-data-eg" class="nav-link px-3 py-1">1. spatial transcriptomics data, e.g.,</a></li>
      <li><a href="#2-single-cell-rnaseq-scrna-seq-data--eg" class="nav-link px-3 py-1">2. single cell RNAseq ((scRNA-seq)) data,  e.g.,</a></li>
    </ul>
  </li>
  <li><a href="#cell-type-deconvolution" class="nav-link px-3 py-1">Cell Type Deconvolution</a>
    <ul>
      <li><a href="#1-create-an-card-object" class="nav-link px-3 py-1">1. Create an <code class="language-plaintext highlighter-rouge">CARD</code> object</a></li>
      <li><a href="#2-deconvolution-using-card" class="nav-link px-3 py-1">2. Deconvolution using CARD</a></li>
      <li><a href="#3-visualize-the-proportion-for-each-cell-type" class="nav-link px-3 py-1">3. Visualize the proportion for each cell type</a></li>
      <li><a href="#4-visualize-the-proportion-for-two-cell-types" class="nav-link px-3 py-1">4. Visualize the proportion for two cell types</a></li>
      <li><a href="#5-visualize-the-cell-type-proportion-correlation" class="nav-link px-3 py-1">5. Visualize the cell type proportion correlation</a></li>
    </ul>
  </li>
  <li><a href="#refined-spatial-map" class="nav-link px-3 py-1">Refined spatial map</a>
    <ul>
      <li><a href="#1-imputation-on-the-newly-grided-spatial-locations" class="nav-link px-3 py-1">1. Imputation on the newly grided spatial locations</a></li>
      <li><a href="#2-visualize-the-cell-type-proportion-at-an-enhanced-resolution" class="nav-link px-3 py-1">2. Visualize the cell type proportion at an enhanced resolution</a></li>
      <li><a href="#3-visualize-the-marker-gene-expression-at-an-enhanced-resolution" class="nav-link px-3 py-1">3. Visualize the marker gene expression at an enhanced resolution</a></li>
    </ul>
  </li>
  <li><a href="#extension-of-card-in-a-reference-free-version-cardfree" class="nav-link px-3 py-1">Extension of CARD in a reference-free version: CARDfree</a>
    <ul>
      <li><a href="#1-create-an-cardfree-object" class="nav-link px-3 py-1">1. Create an <code class="language-plaintext highlighter-rouge">CARDfree</code> object</a></li>
      <li><a href="#2-deconvolution-using-cardfree" class="nav-link px-3 py-1">2. Deconvolution using CARDfree</a></li>
      <li><a href="#3-visualization-of-the-results-of-cardfree" class="nav-link px-3 py-1">3. Visualization of the results of CARDfree</a></li>
    </ul>
  </li>
  <li><a href="#extension-of-card-for-single-cell-resolution-mapping" class="nav-link px-3 py-1">Extension of CARD for single cell resolution mapping</a></li>
</ul>



    <a class="px-3 text-muted" href="#top">
        <small>Back to top</small>
    </a>
</nav>

            
        </aside>
    </div>

    
    <div id="disqus_thread"></div>
    <script>
        var disqus_config = function () {
            this.page.url = 'https://kaizheng-academic.github.io/code/lrtm-2025-tcbb/documentation/04_CARD_Example.html';
            this.page.identifier = 'https://kaizheng-academic.github.io/code/lrtm-2025-tcbb/documentation/04_CARD_Example.html';
        };
        (function() {
            var d = document, s = d.createElement('script');
            s.src = 'https://.disqus.com/embed.js';
            s.setAttribute('data-timestamp', +new Date());
            (d.head || d.body).appendChild(s);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript" rel="nofollow">comments powered by Disqus.</a></noscript>

</div>


        <footer class="site-footer border-top mt-4 py-4 text-center" role="contentinfo">
    

    <div class="container">
    
        <div class="site-footer__social-links">
            <ul class="list-inline mb-2">
                
                    <li class="list-inline-item">
                        <iframe
                            class="github-btn"
                            src="https://ghbtns.com/github-btn.html?user=YMa-lab&amp;repo=CARD&amp;type=watch&amp;count=true"
                            width="90"
                            height="20"
                            title="Star on GitHub"
                        ></iframe>
                    </li>
                    <li class="list-inline-item">
                        <iframe
                            class="github-btn"
                            src="https://ghbtns.com/github-btn.html?user=YMa-lab&amp;repo=CARD&amp;type=fork&amp;count=true"
                            width="90"
                            height="20"
                            title="Fork on GitHub"
                        ></iframe>
                    </li>
                
                
                
                
                
            </ul>
        </div>
    

    <div class="site-footer__body">
        <div class="row align-items-center">
            <div class="col-md-5">
                
                    <ul class="site-footer__links text-left list-inline mb-0">
                        
                            <li class="list-inline-item">
                                <a
                                    href="https://github.com/YMa-lab/CARD/"
                                    class="site-footer__link"
                                >
                                    
    <i
        class="fab fa-github fa-fw "
        aria-hidden="true"
    ></i>


                                    <span>GitHub</span>
                                </a>
                            </li>
                        
                            <li class="list-inline-item">
                                <a
                                    href="https://github.com/YMa-lab/CARD/issues?state=open"
                                    class="site-footer__link"
                                >
                                    
    <i
        class="fas fa-bug fa-fw "
        aria-hidden="true"
    ></i>


                                    <span>Issues</span>
                                </a>
                            </li>
                        
                    </ul>
                
            </div>

            <div class="col-md-7 text-left text-md-right">
                
                    <div class="site-footer__summary">
                        
                        

                        <p class="mb-0">Licensed under

                                <a href="https://www.gnu.org/licenses/licenses.en.html" target="_blank">GNU License 3.0</a></p>

                        <p class="mb-0">
                            Documentation template by <a href="http://getbootstrap.com">Bootstrap team</a>, generated with
                            <a href="https://github.com/allejo/jekyll-docs-theme">Jekyll Docs Theme</a>
                        </p>
                    </div>
                
            </div>
        </div>
    </div>
</div>

    

</footer>



<!-- Bootstrap Related Dependencies -->
<script
    src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.slim.js"
    integrity="sha256-BTlTdQO9/fascB1drekrDVkaKd9PkwBymMlHOiG+qLI=" crossorigin="anonymous"
></script>
<script
    src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.16.1/umd/popper.min.js"
    integrity="sha256-/ijcOLwFf26xEYAjW75FizKVo5tnTYiQddPZoLUHHZ8=" crossorigin="anonymous"
></script>
<script
    src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.4.1/js/bootstrap.min.js"
    integrity="sha256-WqU1JavFxSAMcLP2WIOI+GB2zWmShMI82mTpLDcqFUg=" crossorigin="anonymous"
></script>



    <script src="https://cdnjs.cloudflare.com/ajax/libs/d3/3.4.11/d3.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/trianglify/0.1.2/trianglify.min.js"></script>
    <script>
        let uiColors = [];

        if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
                uiColors = [
                    "#062a48",
                    "#304e67",
                ];
            } else {
                uiColors = [
                    "#080331",
                    "#673051",
                ];
            }
    </script>


<script src="/code/lrtm-2025-tcbb/assets/js/docs.js"></script>





    <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
    ga('create', 'UA-123456-1', 'auto');
    ga('send', 'pageview');
</script>


    </body>
</html>
